parser_library/test/debugging/debugger_test.cpp

```diff
@@ -253,6 +262,46 @@ bool check_step(
     return true;
 }
 
+namespace {
+void step_over_by(size_t number_of_steps,
+    debugger& d,
+    debug_event_consumer_s_mock& m,
+    std::vector<debugging::stack_frame>& exp_stack_frames,
+    size_t line)
+{
+    exp_stack_frames[0].begin_line = exp_stack_frames[0].end_line = line;
+
+    while (number_of_steps > 0)
+    {
+        d.next();
+        m.wait_for_stopped();
+        number_of_steps--;
+    }
+}
+
+void step_into(debugger& d,
+    debug_event_consumer_s_mock& m,
+    std::vector<debugging::stack_frame>& exp_stack_frames,
+    size_t line,
+    std::string name,
+    debugging::source source)
+{
+    exp_stack_frames.insert(
+        exp_stack_frames.begin(), debugging::stack_frame(line, line, exp_stack_frames[0].id + 1, name, source));
```

Comment:
should this handle the empty case as well, just for consistency? or at least assert non-emptiness?

Reply:
It sure can.